### ISSUE TITLE
500 Encoding Error when viewing commits with unicode characters in branchname

### ISSUE THREAD

**Issue by nudgegoonies** [gitlabhq/gitlabhq#5758]
I created a branch inside the test project called täte as a test and made a commit. I click on the commit in gitlab and get the 500 Encoding error.

This ist the production log:
```
Encoding::CompatibilityError (incompatible character encodings: UTF-8 and ASCII-8BIT):
  app/views/projects/commit/_commit_box.html.haml:49:in `_app_views_projects_commit__commit_box_html_haml__356847883316487170_75490980'
  app/views/projects/commit/show.html.haml:1:in `_app_views_projects_commit_show_html_haml___912579123390075438_75432940'
  app/controllers/application_controller.rb:54:in `set_current_user_for_thread'

  Rendered errors/encoding.html.haml within layouts/errors (3.4ms)
  Rendered layouts/_head.html.haml (4.2ms)
  Rendered layouts/_search.html.haml (99.7ms)
  Rendered layouts/_head_panel.html.haml (111.3ms)
  Rendered layouts/_flash.html.haml (0.4ms)
Completed 500 Internal Server Error in 688.4ms (Views: 127.3ms | ActiveRecord: 2.5ms)
```

This is the output of the check (removed project names [...]):
```
Checking Environment ...

Git configured for git user? ... yes
Has python2? ... yes
python2 is supported version? ... yes

Checking Environment ... Finished

Checking GitLab Shell ...

GitLab Shell version >= 1.7.9 ? ... OK (1.8.0)
Repo base directory exists? ... yes
Repo base directory is a symlink? ... no
Repo base owned by git:git? ... yes
Repo base access is drwxrws---? ... yes
update hook up-to-date? ... yes
update hooks in repos are links: ... 
[...]
Running /var/lib/gitlab/gitlab-shell/bin/check
Check GitLab API access: OK
Check directories and files: 
	/var/lib/gitlab/repositories: OK
	/var/lib/gitlab/.ssh/authorized_keys: OK
Test redis-cli executable: redis-cli 2.4.15
Send ping to redis server: PONG
gitlab-shell self-check successful

Checking GitLab Shell ... Finished

Checking Sidekiq ...

Running? ... yes
Number of Sidekiq processes ... 1

Checking Sidekiq ... Finished

Checking GitLab ...

Database config exists? ... yes
Database is SQLite ... no
All migrations up? ... yes
GitLab config exists? ... yes
GitLab config outdated? ... no
Log directory writable? ... yes
Tmp directory writable? ... yes
Init script exists? ... yes
Init script up-to-date? ... yes
projects have namespace: ... 
[...]
Redis version >= 2.0.0? ... yes
Your git bin path is "/usr/bin/git"
Git version >= 1.7.10 ? ... yes (1.7.10)

Checking GitLab ... Finished

```

The gitlab version used is 6.3.0 and the gitlab-shell version used is 1.8.0

P.S.
I tried 6.4.0.pre2 in a VM with our live data with the same result.

**Comment by gzpbx** [comment]
This issue (unicode in a commit ) is stil present in GitLab 7.6.2 16f0968

**Comment by kgoedert** [comment]
same problem with 7.6.2

**Comment by UnderGreen** [comment]
I have same issue, when I try to use blame on commit with '*_ASSERT' on commit message
blame/3d4afd29972e59c3918f5d30388cbbaf184e20d5/Projects/UniSearch/BitField.h
```
Encoding::CompatibilityError (incompatible character encodings: UTF-8 and ASCII-8BIT):
  app/views/projects/blame/show.html.haml:48:in `block in _app_views_projects_blame_show_html_haml__4401936396475562449_27793120'
  app/views/projects/blame/show.html.haml:13:in `_app_views_projects_blame_show_html_haml__4401936396475562449_27793120'
```

**Comment by h3** [comment]
This is rather a critical (or very annoying to say the least) issue for anybody living in a non-English country.

Yet after a year there is still no fix in sight, only pointless workaround..

I find this rather embarrassing..

**Comment by maxlazio** [comment]
@h3 It is unfortunate that you find that embarrassing. Encoding issues are extremely hard to resolve and they take a lot of time. In past I've spent days fixing problems only for that to create further issues. That is why we welcome all contributions which would solve a specific issue while still not breaking everything for everyone.

**Comment by h3** [comment]
Care to elaborate ? I'm genuinely interested to know more about the problem and its side effects.

I know next to nothing about Ruby (I'm more Python), otherwise I would already have tried to fix it..  But a different perspective could maybe lead somewhere.

Cheers

**Comment by nfantone** [comment]
+1 on this. Still happening in 7.7.2.

A colleague tried to push a branch named  `501-rec-editar-campaña`, which wasn't showing up in GitLab.

**Comment by lorenzhs** [comment]
@nfantone in 7.8.2 I can push a branch with that name, but I still get the error from https://github.com/gitlabhq/gitlabhq/issues/5758#issuecomment-41268843 when doing so:
```
%  git push -u origin 501-rec-editar-campaña
Total 0 (delta 0), reused 0 (delta 0)
remote: /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/json-1.8.2/lib/json/common.rb:223:in `encode': "\xC3" on US-ASCII (Encoding::InvalidByteSequenceError)
remote: 	from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/json-1.8.2/lib/json/common.rb:223:in `generate'
remote: 	from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/json-1.8.2/lib/json/common.rb:223:in `generate'
remote: 	from /opt/gitlab/embedded/service/gem/ruby/2.1.0/gems/json-1.8.2/lib/json/common.rb:394:in `dump'
remote: 	from /opt/gitlab/embedded/service/gitlab-shell/lib/gitlab_post_receive.rb:74:in `update_redis'
remote: 	from /opt/gitlab/embedded/service/gitlab-shell/lib/gitlab_post_receive.rb:19:in `exec'
remote: 	from hooks/post-receive:17:in `<main>'
To https://git.foo/bar/baz.git
 * [new branch]      501-rec-editar-campaña -> 501-rec-editar-campaña
Branch 501-rec-editar-campaña set up to track remote branch 501-rec-editar-campaña from origin.
```
Screenshot of the branches list:
![](http://i.imgur.com/OnCRibC.png)

**Comment by nfantone** [comment]
@lorenzhs That certainly does not happen in 7.7.2. We might as well consider upgrading.

PS.: Very nice photo editing skills there.

**Comment by DouweM** [comment]
A fix to this issue has been merged and will be in version 7.9. Please reopen this issue if the problem persists after upgrading to 7.9.
